server/core/src/https/middleware/csp_headers.rs

```diff
@@ -0,0 +1,33 @@
+use axum::extract::State;
+use axum::http::Request;
+use axum::middleware::Next;
+use axum::response::Response;
+
+use crate::https::ServerState;
+
+pub async fn cspheaders_layer<B>(
+    State(state): State<ServerState>,
+    request: Request<B>,
+    next: Next<B>,
+) -> Response {
+    // wait for the middleware to come back
+    let mut response = next.run(request).await;
+
+    // add the header
+    let headers = response.headers_mut();
+    headers.insert("Content-Security-Policy", state.csp_header);
+
+    response
+}
+
+/// Removes the CSP headers from the response
+pub async fn strip_csp_headers<B>(request: Request<B>, next: Next<B>) -> Response {
```

Comment:
Why do we have a "strip" csp header? Shouldn't we just "not add them" ?

Reply:
That was another failed experiment, I just rolled with "CSP all the things".